src/descriptor/mod.rs

```diff
@@ -67,6 +73,228 @@ pub enum Descriptor<Pk: MiniscriptKey> {
     ShWsh(Miniscript<Pk>),
 }
 
+#[derive(Debug, Eq, PartialEq, Clone, Ord, PartialOrd, Hash)]
+pub enum DescriptorKey {
+    PukKey(bitcoin::PublicKey),
+    XPub(DescriptorXPub),
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, Ord, PartialOrd, Hash)]
+pub struct DescriptorXPub {
+    source: Option<([u8; 4], DerivationPath)>,
```

Comment:
All key types seem to allow a source description. Move this to a `DescriptorKey` struct.

Reply:
Might be preferable to use `use bitcoin::util::bip32::Fingerprint` instead of `[u8; 4]`